src/colorizer/FrameCache.ts

```diff
@@ -1,48 +1,54 @@
-import { Texture } from "three";
-
-type CacheEntry = {
-  frame: Texture;
-  index: number;
-  prev: CacheEntryNullable;
-  next: CacheEntryNullable;
+type DisposableValue = {
+  dispose?: () => void;
 };
 
-type CacheEntryNullable = CacheEntry | null;
+type CacheEntry<E extends DisposableValue> = {
+  key: string;
+  value: E;
+  size: number;
+  prev: CacheEntry<E> | null;
+  next: CacheEntry<E> | null;
+};
 
-/** LRU cache for textures. Calls `dispose` on eviction to keep GPU memory under control. */
-export default class FrameCache {
-  private data: CacheEntryNullable[];
-  private first: CacheEntryNullable;
-  private last: CacheEntryNullable;
-  private numItems: number;
+/**
+ * Generic LRU cache for data, intended for Texture or other GPU resources.
+ * Calls `dispose` on eviction to keep GPU memory under control.
+ */
+export default class DataCache<E extends DisposableValue> {
+  private data: Map<string, CacheEntry<E>>;
+  private first: CacheEntry<E> | null;
+  private last: CacheEntry<E> | null;
+  private currentSize: number;
   private maxSize: number;
 
-  constructor(length: number, maxSize: number = 30) {
-    this.data = new Array(length).fill(null);
+  constructor(maxSize: number = 30) {
```

Comment:
shouldn't this maxSize be a byte count?

Reply:
Okay, made a change so the frame cache now has a max limit of 64 MB, based on raw image size. With the default nucmorph dataset, this is about 16 frames. The browser is also doing a lot of caching though, so I'm not confident whether this will have a measurable impact.

Reply:
Also, I've updated some behavior so frames larger than the cache size can still be cached; without this, certain datasets with very large frames (like the endothelial flow datasets, which have a raw frame size of 100+ MB) exceed the cache size.